pkg/lease/manager.go

```diff
@@ -0,0 +1,205 @@
+package lease
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	coordv1 "k8s.io/api/coordination/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apitypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Manager interface {
+	CreateOrGetLease(ctx context.Context, node *corev1.Node, duration time.Duration, holderIdentity string, namespace string) (*coordv1.Lease, bool, error)
+	UpdateLease(ctx context.Context, node *corev1.Node, lease *coordv1.Lease, currentTime *metav1.MicroTime, leaseDuration, leaseDeadline time.Duration, holderIdentity string) (bool, error)
+	InvalidateLease(ctx context.Context, nodeName string, leaseNamespace string) error
```

Comment:
this needs to have the object we create the lease for as arg, not the just name, like in the create method IMHO (at least when we go the generic way)

Reply:
Good chance that I've missed something since I ended up just using the name.
Let me know if you can point me in the correct direction.

Reply:
related to the comment about the name of the lease (adding the object kind for uniqueness): when we delete the lease, we also need both kind and name

Reply:
Putting aside the question whether InvalidateLease should delete or update the Lease, IIUC kind will be needed in any case either for Get or Delete, right ?

Reply:
yes, that's why I proposed to use client.Object here like in the other methods